software/runtime/alloc.c

```diff
@@ -117,11 +117,14 @@ static void *allocate_memory(alloc_t *alloc, const uint32_t size) {
 }
 
 void *domain_malloc(alloc_t *alloc, const uint32_t size) {
-  // Calculate actual required block size (+ size/metadata + alignment)
+  // Calculate actually required block size (+ size/metadata + alignment)
+  // TODO: What's this +4?
```

Comment:
The `+4` is the word for the additional size/metadata field. It could be `sizeof(uint32_t)`.